WordPress/Classes/Networking/RemoteBlogOptionsHelper.m

```diff
@@ -0,0 +1,47 @@
+#import "RemoteBlogOptionsHelper.h"
+
+@implementation RemoteBlogOptionsHelper
+
+// Formats blog options retrieved from REST queries
++ (NSDictionary *)mapOptionsFromResponse:(NSDictionary *)response
+{
+    NSMutableDictionary *options = [NSMutableDictionary dictionary];
+    options[@"home_url"] = response[@"URL"];
+    // We'd be better off saving this as a BOOL property on Blog, but let's do what XML-RPC does for now
+    options[@"blog_public"] = [[response numberForKey:@"is_private"] boolValue] ? @"-1" : @"0";
+    if ([[response numberForKey:@"jetpack"] boolValue]) {
+        options[@"jetpack_client_id"] = [response numberForKey:@"ID"];
+    }
+    if ( response[@"options"] ) {
+        options[@"post_thumbnail"] = [response valueForKeyPath:@"options.featured_images_enabled"];
+        NSArray *optionsDirectMapKeys = @[
+                                          @"active_modules",
+                                          @"admin_url",
+                                          @"login_url",
+                                          @"image_default_link_type",
+                                          @"software_version",
+                                          @"videopress_enabled",
+                                          @"timezone",
+                                          @"gmt_offset",
+                                          @"allowed_file_types",
+                                          ];
+
+        for (NSString *key in optionsDirectMapKeys) {
+            NSString *sourceKeyPath = [NSString stringWithFormat:@"options.%@", key];
+            if ([response valueForKeyPath:sourceKeyPath] != nil) {
+                options[key] = [response valueForKeyPath:sourceKeyPath];
+            }
+        }
+    } else {
+        //valid default values
+        options[@"software_version"] = @"3.6";
```

Comment:
I can almost see the puzzled look on some poor user's face when the app asks them to upgrade the WordPress version in WordPress.com. I wonder if `response[@"options"] == nil` should be treated as an error